sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/application-offering-change-request.institutions.controller.ts

```diff
@@ -0,0 +1,62 @@
+import { Controller, Get, Param, ParseIntPipe, Query } from "@nestjs/common";
+import { ApiTags } from "@nestjs/swagger";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { AllowAuthorizedParty, HasLocationAccess } from "../../auth/decorators";
+import { ClientTypeBaseRoute } from "../../types";
+import { getUserFullName } from "../../utilities";
+import {
+  OfferinfChangePaginationOptionsAPIInDTO,
+  PaginatedResultsAPIOutDTO,
+} from "../models/pagination.dto";
+import BaseController from "../BaseController";
+import { ApplicationOfferingChangeSummaryAPIOutDTO } from "./models/application-offering-change-request.institutions.dto";
+import { ApplicationOfferingChangeRequestService } from "../../services";
+
+/**
+ * Application offering change request controller for institutions client.
+ */
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@Controller("location/:locationId/application-offering-change-request")
+@ApiTags(`${ClientTypeBaseRoute.Institution}-location-application-offering-change-request`)
+export class ApplicationOfferingChangeRequestInstitutionsController extends BaseController {
+  constructor(private readonly applicationOfferingChangeRequestService: ApplicationOfferingChangeRequestService) {
+    super();
+  }
+
+  /**
+   * Gets all eligible applications that can be requested for application
+   * offering change.
+   * @param locationId location id.
+   * @param paginationOptions options to execute the pagination.
+   * @returns list of eligible applications that can be requested for
+   * application offering change.
+   */
+  @HasLocationAccess("locationId")
```

Comment:
Since we have the `location/:locationId` at the controller level, I would prefer to have the decorator move to the class level also.